src/ingest-pipeline/misc/tools/sterilize_fastq.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+
+import gzip
+import random
+import string
+import sys
+
+from argparse import ArgumentParser
+from typing import TextIO
+
+ALPHABETIC_CHARS = string.ascii_uppercase
+NUMERIC_CHARS = string.digits
+RNA_CHARS = 'ACGT'
+QUALITY_CHARS = string.digits + string.ascii_letters + string.punctuation
+
+
+def _get_random_alpha() -> str:
+    return random.choice(ALPHABETIC_CHARS)
+
+
+def _get_random_digit() -> str:
+    return random.choice(NUMERIC_CHARS)
+
+
+def _get_random_rna() -> str:
+    return random.choice(RNA_CHARS)
+
+
+def _get_random_quality() -> str:
+    return random.choice(QUALITY_CHARS)
+
+
+def _get_random_matching_character(input_char) -> str:
+    """This selects a random character, matching the input type.
+
+    * Letters get replaced with random, uppercase letters;
+    * Digits get replaced with random digits; and
+    * all other characters are copied (they may be significant).
+    """
+    return (
+        _get_random_alpha() if str.isalpha(input_char)
+        else _get_random_digit() if str.isdigit(input_char)
+        else input_char
+    )
+
+
+def _open_stream(filename: str, for_writing: bool) -> TextIO:
+    open_mode = 'wt' if for_writing else 'rt'
+
+    if not filename:
+        assert for_writing, 'Only the output filename may be omitted.'
```

Comment:
'if not filename' will indeed work here, but I think it's considered to be more explicit with 'if filename is None'.  If the value of filename was the boolean False (which I admit is unlikely), 'if not filename' does not behave in the intended way.